File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_datetime/DateTimeSelection.java
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_datetime;

import com.github.rooneyandshadows.java.commons.date.DateUtilsOffsetDate;
import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection;

import java.time.OffsetDateTime;

class DateTimeSelection extends BaseDialogSelection<OffsetDateTime> {
    public DateTimeSelection(OffsetDateTime current, OffsetDateTime draft) {
        super(current, draft);
    }

    @Override
    public boolean compareValues(OffsetDateTime v1, OffsetDateTime v2) {
        return DateUtilsOffsetDate.isDateEqual(v1, v2, true);
    }

    @Override
    public boolean hasCurrentSelection() {
        return getCurrentSelection() != null;
    }

    @Override
    public boolean hasDraftSelection() {
        return getDraftSelection() != null;
    }
}

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/picker_dialog_datetime/DateTimeSelection.kt
package com.github.rooneyandshadows.lightbulb.dialogs.picker_dialog_datetime

import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogSelection
import com.github.rooneyandshadows.java.commons.date.DateUtilsOffsetDate
import java.time.OffsetDateTime

internal class DateTimeSelection(current: OffsetDateTime?, draft: OffsetDateTime?) :
    BaseDialogSelection<OffsetDateTime?>(current, draft) {
    override fun compareValues(v1: OffsetDateTime?, v2: OffsetDateTime?): Boolean {
        return DateUtilsOffsetDate.isDateEqual(v1, v2, true)
    }

    override fun hasCurrentSelection(): Boolean {
        return currentSelection != null
    }

    override fun hasDraftSelection(): Boolean {
        return draftSelection != null
    }
}